shell/edit/provisioning.cattle.io.cluster/rke2.vue

```diff
@@ -921,8 +928,24 @@ export default {
       }
     },
 
-    selectedVersion() {
-      this.versionInfo = {}; // Invalidate cache such that version info relevent to selected kube version is updated
+    async selectedVersion(neu, old) {
+      if (this.isEdit) {
+        const { addonConfigDiffs, addonNames, userChartValues } = this;
+        const context = {
+          addonConfigDiffs,
+          addonNames,
+          $store: this.$store,
+          userChartValues
+        };
+
+        await addonConfigPreserve(
+          context,
```

Comment:
This can be simplified to 
```
       const {
          addonConfigDiffs, addonNames, userChartValues, $store
        } = this;

        await addonConfigPreserve(
          {
            addonConfigDiffs, addonNames, userChartValues, $store
          },
          this.originalKubeVersion?.charts,
          neu?.charts
        );
```